common/logstuff.cpp

```diff
@@ -493,7 +493,7 @@ static void InternalLogMail( const char *msgbuffer, unsigned int msglen, int /*f
 // a (va_list *) instead to avoid this problem
 void LogWithPointer( int loggingTo, const char *format, va_list *arglist )
 {
-  char msgbuffer[1024]; //min 1024!!, but also think of other OSs stack!!
+  char msgbuffer[32000]; //min 1024!!, but also think of other OSs stack!!
```

Comment:
How large do you think this buffer actually needs to be? 32k may represent a significant amount of stack space on some older hardware, so we'd like to not unnecessarily enlarge this more than necessary.

Reply:
Very good point, I had to increase this as the BuildCLPrpgram function displayed the log size was 16283 so LogWithPointer failed as it was not going to fit in 1024 bytes. I just set it so it would display the error message. As log messages could be any length.   I would suggest that msgbuffer be allocated on the heap instead of the stack to avoid  possible overrun errors.  I can implement and test to verify functionality.

Reply:
If the message were truncated to 1024 chars (rather than overflowing the buffer and crashing as it presumably did), would that have provided enough information to understand the meaning of the build failure?

Reply:
There was no information provided on the reason for  the build failure. This is the output before I increased the msgbuffer size.

[Feb 26 19:11:55 UTC] Error building cl program on device 2
[Feb 26 19:11:55 UTC] Error code -11, message: Program build failure
[Feb 26 19:11:55 UTC] Build log returned 16263 bytes
Build Log: